pkg/objectfile/cache.go

```diff
@@ -80,5 +81,11 @@ func fromProcess(pid int, m *profile.Mapping) (*MappedObjectFile, error) {
 }
 
 func cacheKey(pid int, m *profile.Mapping) string {
-	return path.Join("/proc", strconv.FormatInt(int64(pid), 10), m.BuildID)
+	// use all filed needed in MappedObjectFile.computeBase to build a unique key
+	b := make([]byte, 0, 3*binary.MaxVarintLen64)
```

Comment:
If we're allocating this anyway, might as well just allocate `3*8+len(m.BuildID)`.